runtime/util/shchelp_j9.c

```diff
@@ -86,3 +86,41 @@ getOpenJ9Sha()
 
 	return sha;
 }
+
+/**
+ * If the class is a lambda class get the pointer to the last '$' sign of the class name which is in the format of HostClassName$$Lambda$<IndexNumber>/0000000000000000.
+ * NULL otherwise.
+ *
+ * @param[in] className  pointer to the class name
+ * @param[in] classNameLength  length of the class name
+ * @return Pointer to the last '$' sign of the class name if it is a lambda class.
+ * 		   NULL otherwise.
+ */
+char*
+getLastDollarSignOfLambdaClassName(const char *className, UDATA classNameLength)
+{
+	if (NULL == className || 0 == classNameLength) {
+		return NULL;
+	}
+
+	/* Get the pointer to the last '$' sign */
+	char* end = strnrchrHelper(className, '$', classNameLength);
+
+	if (NULL != end && (end - 8) - className >= 0) {
```

Comment:
Please bracket the sub-expressions here, in the if statement above, and the one below.
`if ((NULL != end) && ((end - 8) - className >= 0))`